exporter/prometheusremotewriteexporter/wal.go

```diff
@@ -0,0 +1,247 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusremotewriteexporter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/tidwall/wal"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+type walConfig struct {
+	Dir             string        `mapstructure:"path"`
+	CacheSize       int           `mapstructure:"cache-size"`
+	RefreshDuration time.Duration `mapstructure:"truncate-frequency"`
+}
+
+func (wc *walConfig) createWAL() (*wal.Log, error) {
+	if wc == nil {
+		// There are cases for which the WAL can be disabled.
+		// TODO: Perhaps log that the WAL wasn't enabled.
+		return nil, nil
+	}
+	wal, err := wal.Open(filepath.Join(wc.Dir, "prom_rw"), &wal.Options{
+		SegmentCacheSize: wc.segmentCacheSize(),
+		NoCopy:           true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("prometheusremotewriteexporter: failed to open WAL: %w", err)
+	}
+	return wal, nil
+}
+
+func (wc *walConfig) segmentCacheSize() int {
+	if wc == nil {
+		return 0
+	}
+	if wc.CacheSize <= 0 {
+		return 300
+	}
+	return wc.CacheSize
+}
+
+func (wc *walConfig) refreshDuration() time.Duration {
+	if wc == nil {
+		return 0
+	}
+	if wc.RefreshDuration <= 0 {
+		// Following what Prometheus's WAL defaults.
+		return 1 * time.Minute
+	}
+	return wc.RefreshDuration
+}
+
+func (prwe *PrwExporter) walEnabled() bool { return prwe.walConfig != nil }
+
+var errWALDisabled = errors.New("Write-Ahead-Log is disabled")
```

Comment:
Better to use WAL abbreviation as that's what we use everywhere else, including README.
```suggestion
var errWALDisabled = errors.New("WAL is disabled")
```

Reply:
Done, thank you!